### PR TITLE
Omar's Submission

### PR DESCRIPTION
**Pull request by alsyoufomar**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,45 +1,91 @@
...
+  }
+  const total = cart.reduce((acc, curr) => acc + (curr.price * curr.quantity), 0)
 
   return (
     <>
       <header id="store">
         <h1>Greengrocers</h1>
         <ul className="item-list store--item-list">
-          {/* Wrtite some code here... */}
+          <Store
```
Comment (marianphillips) [inline src/App.js:42]: I like that you have done this as a form and have reduced both your filter values to a single state. NIIIIIIIIICE

File: src/App.js [hunk 2]
```diff
@@ -1,45 +1,91 @@
...
+    setForm(x => { return { ...x, [name]: value } })
+  }
+  function addToCart (item) {
+    checkQuantity(item.id) ? IncQuantity(item) : setCart([...cart, { ...item }])
+  }
+  function removeFromCart (item) {
+    item.quantity === 1 ? setCart(x => x.filter(c => c.id !== item.id)) : DecQuantity(item)
+  }
+  function DecQuantity (item) {
+    setCart(x => x.map(c => c.id === item.id ? { ...c, quantity: c.quantity - 1 } : c))
```
Comment (marianphillips) [inline src/App.js:27]: what is c?

File: src/store-items.js [hunk 3]
```diff
@@ -1,54 +1,74 @@
...
+    quantity: 1,
+    type: 'vegetables'
   },
   {
     id: "003-apple",
     name: "apple",
-    price: 0.35
+    price: 0.95,
+    quantity: 1,
+    type: 'fruite'
```
Comment (marianphillips) [inline src/store-items.js:22]: not code related, but fruit does not have an e at the end

File: src/components/cart.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
```
Comment (marianphillips) [inline src/components/cart.js:1]: general comment: I don't know how you've managed to achieve so much of the criteria and extension criteria in such a succinct and clean way! Its amazing!

File: src/App.js [hunk 5]
```diff
@@ -1,45 +1,77 @@
...
-What should a cart item look like? 🤔
-*/
+export default function App () {
+  // Setup state here...
+  const [store, setStore] = useState(initialStoreItems)
+  const [cart, setCart] = useState([])
 
-console.log(initialStoreItems)
+  function addToCart (item) {
+    checkQuantity(item.id) ? IncQuantity(item) : setCart([...cart, { ...item }])
```
Comment (nicopicchio) [inline src/App.js:21]: like the use of the ternary operator. everyone seems to use it apart from me! Need to look into this

File: src/components/cart.js (hunk 4 above)
Comment (alsyoufomar) [reply src/components/cart.js:1]: Thank u so much, Marian!

File: src/App.js (hunk 2 above)
Comment (alsyoufomar) [reply src/App.js:27]: It is a nice variable

**Pull request closed without merging**